lighthouse-core/audits/redirects.js

```diff
@@ -33,38 +33,48 @@ class Redirects extends Audit {
       .then(mainResource => {
         // redirects is only available when redirects happens
         const redirectRequests = Array.from(mainResource.redirects || []);
+
         // add main resource to redirectRequests so we can use it to calculate wastedMs
         redirectRequests.push(mainResource);
-        let totalWastedMs = 0;
 
+        let totalWastedMs = 0;
         const pageRedirects = [];
-        for (let i = 1; i < redirectRequests.length; i++) {
-          const request = redirectRequests[i - 1];
-          const nextRequest = redirectRequests[i];
-          const wastedMs = (nextRequest.startTime - request.startTime) * 1000;
 
-          // We skip the first redirect in our calculations but show it in the table below.
-          // We allow 1 redirect (www. => m.)
-          if (i > 1) {
-            totalWastedMs += wastedMs;
+        for (let i = 0; i < redirectRequests.length; i++) {
```

Comment:
the `if (!initialRequest)` part of this ends up being a little confusing. What about pulling it out, something like


```js
let totalWastedMs = 0;
const pageRedirects = [];

if (redirectRequests.length > 1) {
  pageRedirects.push({
    url: `(Initial: ${redirectRequests[0].url})`,
    wastedMs: 'n/a',
  });
}

for (let i = 1; i < redirectRequests.length; i++) {
  const initialRequest = redirectRequests[i - 1];
  const redirectedRequest = redirectRequests[i];

  const wastedMs = (redirectedRequest.startTime - initialRequest.startTime) * 1000;
  totalWastedMs += wastedMs;

  pageRedirects.push({
    url: redirectedRequest.url,
    wastedMs: Util.formatMilliseconds(wastedMs, 1),
  });
}
```
(I believe that should be equivalent)

Reply:
+1

Reply:
sg thx